gapic/src/main/com/google/gapid/models/Capture.java

```diff
@@ -129,6 +130,57 @@ protected void onUiThreadError(GapisInitException error) {
     });
   }
 
+  public void saveCapture(File file) {
+    LOG.log(INFO, "Saving capture " + file + "...");
+    name = file.getName();
+
+    // TODO: refactor out duplicate code from loadCapture.
+    String canonicalPath;
+    try {
+      File canonicalFile = file.getCanonicalFile();
+      canonicalPath = canonicalFile.getAbsolutePath();
+      if (canonicalFile.getParentFile() != null) {
+        settings.lastOpenDir = canonicalFile.getParentFile().getAbsolutePath();
+      }
+    } catch (IOException e) {
+      if (file.getParentFile() != null) {
+        settings.lastOpenDir = file.getParentFile().getAbsolutePath();
+      }
+
+      fireError(new GapisInitException(
+          GapisInitException.MESSAGE_TRACE_FILE_LOAD_FAILED + file, "Loading trace failed", e));
+      return;
+    }
+
+    settings.addToRecent(canonicalPath);
+
+    Rpc.listen(client.exportCapture(path), rpcController,
+        new UiErrorCallback<byte[], Boolean, String>(shell, LOG) {
+          @Override
+          protected ResultOrError<Boolean, String> onRpcThread(Result<byte[]> result) throws RpcException, ExecutionException {
+            try {
+              byte[] data = result.get();
+              try (FileOutputStream fos = new FileOutputStream(file)) {
+                fos.write(data);
+              }
+              return success(true);
+            } catch (ExecutionException | RpcException | IOException e) {
+              return error(e.toString());
```

Comment:
Change the error type to be Exception and then log the exception in the error handler.